### ISSUE TITLE
VS crashes when searching for files in the Solution Explorer

### ISSUE THREAD

**Issue by vritant24** [dotnet/project-system#6432]
**Visual Studio Version**:
Version 16.8.0 Preview 1.0 [30320.23.master]

**Summary**:
Visual Studio Crashes when I try to search for a file in the Solution Explorer

**Steps to Reproduce**:

1. Open/create a project with files in it.
2. in the solution explorer type in a file extension. ".cs" or ".xaml" for instance

**Expected Behavior**:
List matching items in the solution explorer

**Actual Behavior**:
It lists the items, but proceeds to crash VS consistently

**Comment by davkean** [comment]
@vritant24 I can't repro this - can you look at your Event Viewer -> Adminstrative Events and find the crash and grab the call stack?

**Comment by davkean** [comment]
nevermind, I repro'd it:

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentException
   at Microsoft.VisualStudio.PlatformUI.HierarchyItem.CompareTo(System.Object)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections.AggregateRelationCollectionSource+PrioritizedComparableComparer.Compare(System.Object, System.Object)
   at System.Array+SorterObjectArray.InsertionSort(Int32, Int32)
   at System.Array+SorterObjectArray.IntroSort(Int32, Int32, Int32)
   at System.Array+SorterObjectArray.IntrospectiveSort(Int32, Int32)

Exception Info: System.InvalidOperationException
   at System.Array+SorterObjectArray.IntrospectiveSort(Int32, Int32)
   at System.Array.Sort(System.Array, System.Array, Int32, Int32, System.Collections.IComparer)
   at System.Collections.ArrayList.Sort(Int32, Int32, System.Collections.IComparer)
   at System.Collections.ArrayList.Sort(System.Collections.IComparer)
   at MS.Internal.Data.SortFieldComparer.SortHelper(System.Collections.ArrayList, System.Collections.IComparer)
   at MS.Internal.Data.DataExtensionMethods.Sort(System.Collections.IList, System.Collections.IComparer)
   at System.Windows.Data.ListCollectionView.PrepareLocalArray()
   at System.Windows.Data.ListCollectionView.RefreshOverride()
   at System.Windows.Data.CollectionView.RefreshInternal()
   at System.Windows.Data.CollectionView.RefreshOrDefer()
   at System.Windows.Data.ListCollectionView.set_CustomSort(System.Collections.IComparer)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections.AggregateRelationCollectionSource.SetCollection(Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections.IAggregateRelationCollection)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections.RelationAttachedCollectionSourceProviderBase.CreateCollectionSource(System.Object, System.String)
   at Microsoft.VisualStudio.PlatformUI.AttachedCollectionService+<>c__DisplayClass19_0.<AggregateItemsSource>b__0(System.Lazy`2<Microsoft.VisualStudio.Shell.IAttachedCollectionSourceProvider,Microsoft.VisualStudio.Utilities.IOrderable>)
   at System.Linq.Enumerable+WhereSelectListIterator`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Linq.Enumerable+WhereEnumerableIterator`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Collections.Generic.List`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]..ctor(System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at Microsoft.VisualStudio.PlatformUI.AttachedCollectionService.AggregateItemsSource(System.Object, System.String)
   at Microsoft.VisualStudio.PlatformUI.AttachedCollectionService.GetOrCreateCollectionSource(System.Object, System.String)
   at Microsoft.VisualStudio.PlatformUI.SolutionSearchNavigationEntry+<>c.<OnNavigateTo>b__27_0(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetContainersFromItem(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.<GetNodesForFilter>b__113_0(System.Object)
   at System.Linq.Enumerable+<SelectManyIterator>d__17`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.<GetNodesForFilter>b__113_0(System.Object)
   at System.Linq.Enumerable+<SelectManyIterator>d__17`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.<GetNodesForFilter>b__113_0(System.Object)
   at System.Linq.Enumerable+<SelectManyIterator>d__17`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.<GetNodesForFilter>b__113_0(System.Object)
   at System.Linq.Enumerable+<SelectManyIterator>d__17`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.<GetNodesForFilter>b__113_0(System.Object)
   at System.Linq.Enumerable+<SelectManyIterator>d__17`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.GetNodesForFilter(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.AddFilteredItems(System.Collections.IEnumerable)
   at Microsoft.Internal.VisualStudio.PlatformUI.VirtualizingTreeView.OnFilteredItemsChanged(System.Object, System.Collections.Specialized.NotifyCollectionChangedEventArgs)
   at System.Windows.WeakEventManager+ListenerList`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].DeliverEvent(System.Object, System.EventArgs, System.Type)
   at System.Windows.WeakEventManager.DeliverEvent(System.Object, System.EventArgs)
   at System.Collections.Specialized.CollectionChangedEventManager.OnCollectionChanged(System.Object, System.Collections.Specialized.NotifyCollectionChangedEventArgs)
   at Microsoft.VisualStudio.PlatformUI.ExtensionMethods.RaiseEvent(System.Collections.Specialized.NotifyCollectionChangedEventHandler, System.Object, System.Collections.Specialized.NotifyCollectionChangedEventArgs)
   at Microsoft.VisualStudio.PlatformUI.SolutionNavigatorSearchResultSet.AddPrivate(System.Object)
   at Microsoft.VisualStudio.PlatformUI.SolutionNavigatorSearchResultSet.HandleNotification(System.Object)
   at Microsoft.Internal.VisualStudio.PlatformUI.CrossThreadNotificationQueue`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].DrainQueue()
   at Microsoft.Internal.VisualStudio.PlatformUI.CrossThreadNotificationQueue`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Synchronize()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)


```

@drewnoakes This might have been https://github.com/dotnet/project-system/pull/6385?

**Comment by drewnoakes** [comment]
I can't repro this in Version 16.8.0 Preview 1.0 [30323.108.master]

![image](https://user-images.githubusercontent.com/350947/88348736-914e4800-cd91-11ea-8332-6d3d98619f22.png)

> @drewnoakes This might have been #6385?

Looking at the stack I believe it would have been fixed in https://github.com/dotnet/project-system/pull/6365.

**Comment by drewnoakes** [comment]
@davkean what version of VS did you repro this in? My build is a few days newer than @vritant24's, but I would have expected that fix to have been inserted more than a week ago. It did merge into 16.7.x though, which may have take some time to FI into master.

**Comment by davkean** [comment]
I can repro this on: Version 16.8.0 Preview 1.0 [30322.230.master].

**Comment by drewnoakes** [comment]
Thanks, I can repro it.

**Comment by drewnoakes** [comment]
>>@drewnoakes This might have been #6385?
>
>Looking at the stack I believe it would have been fixed in #6365.

You're right, it was introduced in #6385, which is good as that's not in 16.7.
